sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/KalixRunner.scala

```diff
@@ -81,17 +83,40 @@ object KalixRunner {
   def prepareConfig(config: Config): Config = {
 
     val mainConfig = config.getConfig("kalix.system").withFallback(config)
-    val devConfigFile = new File(System.getProperty("user.dir"), "dev-mode.conf")
 
-    Option
-      .when(devConfigFile.exists()) {
-        ConfigFactory.parseFile(devConfigFile)
+    // enrich config with extra dev-mode service port mappings
+    if (mainConfig.hasPath("kalix.dev-mode.service-port-mappings")) {
+
+      val mappings = mainConfig.getConfig("kalix.dev-mode.service-port-mappings")
+      // each kalix.dev-mode.service-port-mappings becomes a new akka.grpc.client service-discovery mapping
+      // that is then appended to the main configuration
+      mappings.entrySet().asScala.foldLeft(mainConfig) { (main, entry) =>
+        val serviceName = entry.getKey
+        val port = entry.getValue.render()
+
+        logger.info(s"Adding dev-mode service port mapping for $serviceName:$port")
+        val mapping = ConfigFactory.parseString(s"""
+             |akka.grpc.client.$serviceName {
+             |  service-discovery {
+             |    service-name = "$serviceName"
+             |  }
+             |  host = "localhost"
```

Comment:
Not sure if this needs to be `0.0.0.0` for cross-platform compatibility... does it? 🤔

Reply:
Good point. I will change to `0.0.0.0`.